### PR TITLE
add "nosuid" flag to exec directory mounts of DynamicUser=1 services

### PR DESCRIPTION
**Pull request by poettering**
Some additional hardening for DynamicUser=1 services.

### CODE REVIEW

**Comment by topimiettinen** [comment]
How about `nodev` and maybe `noexec` too?

File: src/shared/mount-util.c [hunk 1]
```diff
@@ -245,7 +250,7 @@ int bind_remount_recursive_with_mountinfo(const char *prefix, bool ro, char **bl
                         (void) get_mount_flags(cleaned, &orig_flags);
                         orig_flags &= ~MS_RDONLY;
 
-                        if (mount(NULL, cleaned, NULL, orig_flags|MS_BIND|MS_REMOUNT|(ro ? MS_RDONLY : 0), NULL) < 0)
+                        if (mount(NULL, cleaned, NULL, (orig_flags & ~flags_mask)|MS_BIND|MS_REMOUNT|new_flags, NULL) < 0)
```
Comment (keszybz) [inline src/shared/mount-util.c:253]: There's a slight change in behaviour here: before, if the mount was ro, but ro wasn't requested, the mount would remain ro. Now, it'll be remounted rw. But looking at the list of callers, they all request ro, so the point is moot.

**Comment by keszybz** [comment]
I don't think we want `noexec`. `nodev` would seem reasonable. But I guess we can just as well do that later.

**Comment by poettering** [comment]
@topimiettinen nodev probably makes sense for all cases, but the nosuid in particular for DynamicUser= services, since acquiring other identities through suid/sgid is particularly problematic for DynamicUser= services.

But yeah, we might want to add nodev sooner or later too for these mounts, but not conditionalize them on DynamicUser= unlikely nosuid as this PR does.

**Pull request merged**